rust_src/src/editfns.rs

```diff
@@ -0,0 +1,48 @@
+//! Lisp functions pertaining to editing.
+
+use remacs_macros::lisp_fn;
+use lisp::LispObject;
+use remacs_sys::EmacsInt;
+use threads::ThreadState;
+use buffers::get_buffer;
+
+
+/// Return value of point, as an integer.
+/// Beginning of buffer is position (point-min).
+#[lisp_fn]
+pub fn point() -> LispObject {
+    let buffer_ref = ThreadState::current_buffer();
+    LispObject::from_natnum(buffer_ref.pt as EmacsInt)
+}
+
+/// Return the number of characters in the current buffer.
+/// If BUFFER is not nil, return the number of characters in that buffer
+/// instead.
+
+/// This does not take narrowing into account; to count the number of
+/// characters in the accessible portion of the current buffer, use
+/// `(- (point-max) (point-min))', and to count the number of characters
+/// in some other BUFFER, use
+/// `(with-current-buffer BUFFER (- (point-max) (point-min)))'.
+#[lisp_fn(min = "0")]
+pub fn buffer_size(object: LispObject) -> LispObject {
+    let buffer_ref = if object.is_not_nil() {
+        get_buffer(object).as_buffer_or_error()
+    } else {
+        ThreadState::current_buffer()
+    };
+    LispObject::from_natnum((buffer_ref.z() - buffer_ref.beg_byte()) as EmacsInt)
+}
+
+/// Return t if point is at the end of the buffer.
+/// If the buffer is narrowed, this means the end of the narrowed part.
+#[lisp_fn]
+pub fn eobp() -> LispObject {
+    let buffer_ref = ThreadState::current_buffer();
+    let zv = LispObject::from_natnum(buffer_ref.zv() as EmacsInt);
+    if point() == zv {
```

Comment:
This will work since lisp integers should have a unique representation.

Still, better to compare the ptrdiffs directly:
```
LispObject::from_bool(buffer_ref.zv() == buffer_ref.pt)
```